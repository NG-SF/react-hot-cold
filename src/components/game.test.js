import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Game from './game';


describe('<Game />', () => {
  it('Renders without crashing', () => {
    shallow(< Game />);
  });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <Game />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});