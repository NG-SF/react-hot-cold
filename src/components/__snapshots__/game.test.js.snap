// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> should match its empty snapshot 1`] = `
<div>
  <header>
    <nav>
      <ul
        className="clearfix"
      >
        <li>
          <a
            aria-label="How to play"
            className="what"
            href="#what"
          >
            What?
          </a>
        </li>
        <li>
          <a
            aria-label="Start a new game"
            className="new"
            href="#feedback"
            onClick={[Function]}
          >
            + New Game
          </a>
        </li>
        <li>
          <a
            className="visuallyhidden focusable status-link"
            href="#get-status"
            onClick={[Function]}
          >
            Hear state of game
          </a>
        </li>
      </ul>
    </nav>
    <h1>
      HOT or COLD
    </h1>
  </header>
  <main
    role="main"
  >
    <section
      aria-describedby="feedback"
      aria-label="Guess section"
    >
      <h2
        aria-atomic="true"
        aria-live="assertive"
        id="feedback"
        role="status"
      >
        Make your guess!
         
      </h2>
      <form
        onSubmit={[Function]}
      >
        <input
          aria-labelledby="feedback"
          autoComplete="off"
          className="text"
          id="userGuess"
          max="100"
          min="1"
          name="userGuess"
          required={true}
          type="number"
        />
        <button
          className="button"
          id="guessButton"
          name="submit"
          type="submit"
        >
          Guess
        </button>
      </form>
    </section>
    <section
      aria-describedby="guessList"
      aria-labelledby="guessCount"
    >
      <h2
        id="guessCount"
      >
        You've made 
        <span
          id="count"
        >
          0
        </span>
         
        guesses
        !
      </h2>
      <ul
        className="guessBox clearfix"
        id="guessList"
      />
      <p
        aria-atomic="true"
        aria-live="assertive"
        className="visuallyhidden"
        id="status-readout"
      >
        
      </p>
    </section>
    <section
      id="what"
      tabIndex="-1"
    >
      <h2>
        What do I do?
      </h2>
      <p>
        This is a Hot or Cold Number Guessing Game. The game goes like this:
      </p>
      <ol>
        <li>
          I pick a 
          <strong>
            random secret number
          </strong>
           between 1 to 100 and keep it hidden.
        </li>
        <li>
          You need to 
          <strong>
            guess
          </strong>
           until you can find the hidden secret number.
        </li>
        <li>
          You will 
          <strong>
            get feedback
          </strong>
           on how close ("hot") or far ("cold") your guess is.
        </li>
      </ol>
      <p>
        If you're using a screen reader, you can use the visually hidden "Hear state of game" link in the navigation region to get a reminder of all your past guesses and how close your current guess is to being correct.
      </p>
    </section>
  </main>
</div>
`;
